framework/source/class/qx/ui/core/MExecutable.js

```diff
@@ -167,6 +168,7 @@ qx.Mixin.define("qx.ui.core.MExecutable",
             // check also for themed values [BUG #5906]
             if (selfPropertyValue == null) {
               // update the appearance to make sure every themed property is up to date
+              this.$$resyncNeeded = true;
```

Comment:
My objection is that $$resyncNeeded is effectively a private state variable for `Widget` and that by accessing it from outside the `Widget` class you're breaking a fundamental "blackboxing" principal of OO.  

While `MExecutable` is not _really_ suited to be used except in a class derived from `Widget` (although it could be), it is perfectly acceptable for a `Widget` to implement it's own `command` property should it choose to do so - for example, if the developer decided that they disagreed with how properties are bound from the command, they might decide to avoid `MExecutable` altogether.  This would be reasonable because that's how the code is packaged at the moment.  However, now that there is a `$$resyncNeeded` property, the developer would be *forced* to use an internal, private state variable to complete the functionality.

Looking at `Widget` in isolation, `$$resyncNeeded` is largely undocumented, and it's increasingly confusing as to what's going on when there is also `.syncAppearance()` (which is called immediately after `MExecutable` sets `$$resyncNeeded=true`) and also `.updateAppearance()`.

I guess my point is that not having deeply looked at this piece of code before, it's apparent that there are some significant and sometime subtle things going on in `Widget` and IMHO that makes it especially important that behaviours of the `Widget` are very clearly defined and make sense all on their own.